### ISSUE TITLE
Ensemble construction from multi-metric checkpoints

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/vascadia#80]
## Context

The multi-metric tracking system saves best-by-X checkpoints per fold per loss. This enables rich ensemble construction without re-training:
- **Cross-fold:** Combine 3 fold models (same loss) → averaged predictions
- **Cross-loss:** Combine best models from different losses → complementary strengths  
- **Cross-metric:** Load best-by-dice vs best-by-clDice vs best-by-MASD → diverse models
- **Model soup:** Average model weights (Wortsman et al. 2022)

### Tier 1: Prediction-level ensembles
1. Load soft predictions from .npz files (depends on #77)
2. Average/weighted-average soft predictions across ensemble members
3. Apply argmax on averaged predictions
4. Evaluate ensemble with MetricsReloaded

### Tier 2: Weight-level ensembles  
1. Load model state_dicts from multiple checkpoints
2. Model soup: element-wise average of weights
3. Greedy soup: iteratively add models that improve validation metric

### Tier 3: Advanced ensembles
1. Conformal prediction sets from ensemble disagreement
2. Learned ensemble weights via validation performance

## API Design

```python
ensemble = EnsembleBuilder()
ensemble.add_member("runs:/{run_id}/model", weight=0.5)
ensemble.add_member("runs:/{run_id_2}/model", weight=0.5)
predictions = ensemble.predict(volume)
```

## Ref

- User requirement: "flexibly ensemble easily 1) different hyperparameter runs of the same model, 2) different folds of the same model (3-fold CV), 3) totally different models"
- `docs/planning/experiment-planning-and-metrics-prompt.md` — ensembling requirements
